library/common/network/mobile_utility.cc

```diff
@@ -21,6 +53,43 @@ namespace {
 #define SUPPORTS_GETIFADDRS
 #endif
 
+namespace {
+
+// Internal SocketOptionImpl that permutes hash key based on name and value.
+class InternalOptionImpl : public SocketOptionImpl {
```

Comment:
You don't need this anymore as long as you have an envoy that has my retry options predicate PR.

Reply:
Thanks, will bump upstream and update.

Reply:
https://github.com/envoyproxy/envoy-mobile/pull/1806